File: questModule/src/main/java/ru/nekit/android/qls/utils/ActivityUtils.java
package ru.nekit.android.qls.utils;

import android.app.ActivityManager;
import android.content.Context;
import android.support.annotation.NonNull;

/**
 * Created by nekit on 20.02.17.
 */

public class ActivityUtils {

    public static boolean isServiceRunning(@NonNull Context context, Class<?> serviceClass) {
        ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        for (ActivityManager.RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE)) {
            if (serviceClass.getName().equals(service.service.getClassName())) {
                return true;
            }
        }
        return false;
    }
}

File: questPresentation/src/main/java/ru/nekit/android/qls/utils/ActivityUtils.kt
package ru.nekit.android.qls.utils

import android.app.ActivityManager
import android.content.Context

object ActivityUtils {

    fun isServiceRunning(context: Context, serviceClass: Class<*>): Boolean {
        val manager = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        return manager.getRunningServices(Integer.MAX_VALUE).any { serviceClass.name == it.service.className }
    }
}
